### PR TITLE
Fix division by zero

### PR DESCRIPTION
**Pull request by bkentel**
The old code divided by zero -- this should be what was intended?

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -9612,11 +9612,9 @@ void player::rooted()
...
 // Overfiling triggered hibernation checks, so capping.
 {
-    double shoe_factor = footwear_factor();
-    if( (has_trait("ROOTS2") || has_trait("ROOTS3")) &&
-        g->m.has_flag("DIGGABLE", posx(), posy()) &&
-        !shoe_factor ) {
-        if( one_in(20 / shoe_factor) ) {
+    double const shoe_factor = footwear_factor();
+    if ((has_trait("ROOTS2") || has_trait("ROOTS3")) && g->m.has_flag("DIGGABLE", posx(), posy())) {
+        if( one_in(20 * (1 + shoe_factor)) ) {
```
Comment (kevingranade) [inline src/player.cpp]: Not quite what is desired, though I don't the existing code does it either.
Pretty sure what we want is:
No shoes: 1/20 change of feeding.
One shoe: 1/40 chance of feeding
Two shoes: no chance of feeding.

Comment (KA101) [inline src/player.cpp]: Yeah, shoes block rooting.

**Comment by kevingranade** [comment]
Superseded by #11712

**Comment by bkentel** [comment]
Thanks, I'll close this then.

**Pull request closed without merging**